### PR TITLE
add cypress eslint plugin

### PR DESCRIPTION
**Pull request by cescoferraro**
This PR add cypress eslint plugin so we can be aware of cypress related issues

## Changes

- adds eslitn plugin

## Checklist

- [x] tested locally
- [ ] added new dependencies
- [ ] updated the docs
- [ ] added a test

### CODE REVIEW

File: .github/workflows/pull-request.yaml [hunk 1]
```diff
@@ -84,6 +84,10 @@ jobs:
         run: |
           cd web/
           npm test
+      - name: Run eslint
+        run: |
+          cd web/
+          npm run eslint
```
Comment (jorgeepc) [inline .github/workflows/pull-request.yaml:90]: What do you think if we first run the `eslint` and then the `unit tests`? Just to make it fail fast if something does not pass the eslint validation.

File: web/.eslintrc.js [hunk 2]
```diff
@@ -10,8 +11,9 @@ module.exports = {
     ecmaVersion: 12,
     sourceType: 'module',
   },
-  plugins: ['react', '@typescript-eslint', 'react-hooks', 'unused-imports', 'prettier'],
+  plugins: ['react', '@typescript-eslint', 'react-hooks', 'unused-imports', 'prettier', 'cypress'],
   rules: {
+    'cypress/no-unnecessary-waiting': 'error',
```
Comment (jorgeepc) [inline web/.eslintrc.js:16]: Should we use the recommended config?
```
{
  "extends": [
    "plugin:cypress/recommended"
  ]
}
```

File: web/cypress/integration/Trace/CreateAssertion.spec.ts [hunk 3]
```diff
@@ -1,215 +1,182 @@
...
-
-      cy.get('[data-cy=assertion-card-list]').should('be.visible');
-      cy.get('[data-cy=assertion-card]').should('have.lengthOf', 1);
-      deleteTest(testId);
-    })();
+    cy.get(`[data-cy=trace-node-database]`, {timeout: 20000}).last().click();
+    cy.get('[data-cy=add-assertion-button]').click();
+    cy.get('[data-cy=assertion-form]', {timeout: 10000}).should('be.visible');
+
+    // cy.get('[data-cy=mode-selector-switch]').click();
```
Comment (jorgeepc) [inline web/cypress/integration/Trace/CreateAssertion.spec.ts:66]: Should we remove this line?

File: web/cypress/integration/TraceDetails/ShowTestDetails.spec.ts [hunk 4]
```diff
@@ -36,40 +34,46 @@ describe('Show test details', () => {
...
-      deleteTest(testId);
-    })();
+      // const testRunResultId = getResultId(testId);
+      //
+      // cy.get('[data-cy=test-header-back-button]').click();
+      // cy.wait('@testObject');
+      // cy.get(`[data-cy=result-card-${testRunResultId}]`, {timeout: 10000}).should('be.visible');
+      // cy.visit(`http://localhost:3000/test/${testId}/run/${testRunResultId}`);
+      // cy.location('pathname').should('match', testRunPageRegex);
+    });
```
Comment (jorgeepc) [inline web/cypress/integration/TraceDetails/ShowTestDetails.spec.ts:57]: same here

File: web/cypress/support/commands.ts [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+    const localTestId = getTestId(pathname);
+    if (shoudlIntercept) {
+      cy.inteceptHomeApiCall();
+    }
+    cy.visit(`http://localhost:3000`);
+    cy.wait('@testList');
+    cy.get(`[data-cy=test-actions-button-${localTestId}]`).should('be.visible');
+    cy.get(`[data-cy=test-actions-button-${localTestId}]`).click({force: true});
+    cy.get('[data-cy=test-card-delete]').click();
+    // cy.wait('@testDelete');
```
Comment (jorgeepc) [inline web/cypress/support/commands.ts:59]: ?

File: web/package.json [hunk 6]
```diff
@@ -64,9 +64,10 @@
     "types:generate": "openapi-typescript ../api/openapi.yaml --output ./src/types/Generated.types.ts",
     "grammar:generate": "lezer-generator ./src/utils/grammar/syntax.grammar -o ./src/utils/grammar/grammar.js",
     "types": "tsc",
-    "eslint": "eslint ./src/**/*.{ts,tsx}",
+    "eslint": "eslint ./src/**/*.{ts,tsx} ./cypress/**/*.{ts,tsx}",
     "eject": "react-scripts eject",
     "cy:open": "cypress open",
+    "cy:clean": "cypress install",
```
Comment (jorgeepc) [inline web/package.json:70]: Curious, why do we need this?

File: web/cypress/integration/Trace/CreateAssertion.spec.ts (hunk 3 above)
Comment (cescoferraro) [reply web/cypress/integration/Trace/CreateAssertion.spec.ts:66]: we dont have the notion of advanced editor. do know why this is not breaking current test.

File: web/cypress/support/commands.ts (hunk 5 above)
Comment (cescoferraro) [reply web/cypress/support/commands.ts]: now this is uncommented

File: web/cypress/integration/TraceDetails/ShowTestDetails.spec.ts (hunk 4 above)
Comment (cescoferraro) [reply web/cypress/integration/TraceDetails/ShowTestDetails.spec.ts]: now this is uncommented

**Review by xoscar — APPROVED** [review]
LGTM

**Review by jorgeepc — APPROVED** [review]

**Pull request merged**